### ISSUE TITLE
broadcastEval throws SHARDING_IN_PROCESS error even though it should execute no matter how many shards are open

### ISSUE THREAD

**Issue by NobUwU** [discordjs/discord.js#5372]
broadcastEval throws a SHARDING_IN_PROCESS error when trying to use during shards being spawned however this was not an issue in older versions of the library

shard.js
```js
const { ShardingManager } = require('discord.js')
const constants = require('./constants')

const manager = new ShardingManager('./src/index.js', {
  token: process.env.TOKEN,
  totalShards: constants.totalShards,
})

manager.spawn(process.env.TOTAL_SHARDS, 10000, -1)
```
index.js
```js
const { Client } = require('discord.js')
const client = new Client()

client.on('ready', () => {
   client.shard.broadcastEval(`
      this.user.setActivity("Bot Starting Up...", { type: "PLAYING" }) 
   `).then().catch()
})

client.login(process.env.TOKEN)
```
**Further details:**

- discord.js version: ^12.5.1
- Node.js version: 14.15.4
- Operating system: Windows 10

**Comment by iCrawl** [comment]
Define:
> an issue in older versions of the library

If you upgraded from v11 to v12 breaking changes are to be expected.

**Comment by iCrawl** [comment]
Ref: https://github.com/discordjs/discord.js/pull/5222

**Comment by MattIPv4** [comment]
Working around this is relatively easy if you have a need to run `broadcastEval` on your shards whilst some are spawning -- just iterate over each shard and call `broadcastEval` with the shard number, catching the error thrown for any shards that are currently being spawned.

I would not consider `SHARDING_IN_PROCESS` being thrown to be a bug here, you are trying to call `broadcastEval` which is designed by default to broadcast to all shards, but not all shards are available so you are getting the appropriate error back.

**Comment by kyranet** [comment]
Closing as it's resolved in the master branch with #5222.
